src/foundry/common/documents/folder.d.mts

```diff
@@ -29,19 +29,41 @@ declare abstract class BaseFolder<out _SubType extends BaseFolder.SubType = Base
    */
   constructor(...args: Folder.ConstructorArgs);
 
+  /**
+   * @defaultValue
+   * ```js
+   * mergeObject(super.metadata, {
+   *   name: "Folder",
+   *   collection: "folders",
+   *   label: "DOCUMENT.Folder",
+   *   labelPlural: "DOCUMENT.Folders",
+   *   coreTypes: CONST.FOLDER_DOCUMENT_TYPES,
+   *   schemaVersion: "12.324"
+   * })
+   * ```
+   */
   static override metadata: BaseFolder.Metadata;
 
   static override defineSchema(): BaseFolder.Schema;
 
-  // NOTE(LukeAbby): The `validateJoint` method override was left off because it didn't change the signature and it also seemed to cause a loop.
-  // See: https://gist.github.com/LukeAbby/43ee5c2a39cd33f3fac693d1d4a5653f
+  /**
+   * @remarks
+   * @throws If `data.folder === data._id` (no putting folders inside themselves)
+   * */
+  static validateJoint(data: Folder.Source): void;
 
   /**
    * Allow folder sorting modes
    * @defaultValue `["a", "m"]`
    */
   static SORTING_MODES: ("a" | "m")[];
 
+  /**
+   * @remarks Never returns an index entry, only ever {@linkcode Folder.Implementation} or `null`, as the `folders` collection of a
+   * compendium is accessible at sync speed
```

Comment:
Sync "speed"? Do you mean synchronously? Asynchronous vs synchronous isn't a speed, it's a strategy. Async is essentially syntax sugar for callbacks added to the end of the event loop whereas synchronous just starts executing (though you could think of it as callbacks pushed to the start of the queue).